numba/annotations/step.py

```diff
@@ -0,0 +1,143 @@
+"""
+A light and fast template engine.
```

Comment:
These are good points. I was mostly trying to avoid adding a dependency just for some basic templating, but if everyone else is okay with a new dependency then I'm fine with that too.